app/client/components/export-dialog.tsx

```diff
@@ -0,0 +1,460 @@
+	trigger: React.ReactNode;
+	variant?: never;
+	size?: never;
+	triggerLabel?: never;
+	showIcon?: never;
+};
+
+type ExportDialogWithDefaultTrigger = BaseExportDialogProps & {
+	trigger?: never;
+	variant?: "default" | "outline" | "ghost" | "card";
+	size?: "default" | "sm" | "lg" | "icon";
+	triggerLabel?: string;
+	showIcon?: boolean;
+};
+
+type ExportDialogProps = ExportDialogWithTrigger | ExportDialogWithDefaultTrigger;
+
+export function ExportDialog({
+	entityType,
+	name,
+	id,
+	trigger,
+	variant = "outline",
+	size = "default",
+	triggerLabel,
+	showIcon = true,
+}: ExportDialogProps) {
+	const [open, setOpen] = useState(false);
+	const [includeIds, setIncludeIds] = useState(true);
+	const [includeTimestamps, setIncludeTimestamps] = useState(true);
+	const [includeRuntimeState, setIncludeRuntimeState] = useState(false);
+	const [includeRecoveryKey, setIncludeRecoveryKey] = useState(false);
+	const [includePasswordHash, setIncludePasswordHash] = useState(false);
+	const [secretsMode, setSecretsMode] = useState<SecretsMode>("exclude");
+	const [isExporting, setIsExporting] = useState(false);
+	const [showPasswordStep, setShowPasswordStep] = useState(false);
+	const [password, setPassword] = useState("");
+	const [isVerifying, setIsVerifying] = useState(false);
+
+	const config = exportConfigs[entityType];
+	const isSingleItem = !!(name || id);
+	const isFullExport = entityType === "full";
+	// TODO: Volumes will have encrypted secrets (e.g., SMB/NFS credentials) in a future PR
+	const hasSecrets = entityType !== "backup-schedules" && entityType !== "volumes";
+	const entityLabel = isSingleItem ? config.label : config.labelPlural;
+	const requiresPassword = includeRecoveryKey || secretsMode === "cleartext";
+
+	const performExport = async () => {
+		setIsExporting(true);
+		try {
+			await exportConfig(entityType, {
+				includeIds,
+				includeTimestamps,
+				includeRuntimeState,
+				includeRecoveryKey: isFullExport ? includeRecoveryKey : undefined,
+				includePasswordHash: isFullExport ? includePasswordHash : undefined,
+				secretsMode: hasSecrets ? secretsMode : undefined,
+				name,
+				id,
+			});
+			toast.success(`${entityLabel} exported successfully`);
+			setOpen(false);
+			setShowPasswordStep(false);
+			setPassword("");
+		} catch (err) {
+			toast.error("Export failed", {
+				description: err instanceof Error ? err.message : String(err),
+			});
+		} finally {
+			setIsExporting(false);
+		}
+	};
+
+	const handleExport = () => {
+		if (requiresPassword) {
+			setShowPasswordStep(true);
+		} else {
+			performExport();
+		}
+	};
+
+	const handlePasswordSubmit = async (e: React.FormEvent) => {
+		e.preventDefault();
+		if (!password) {
+			toast.error("Password is required");
+			return;
+		}
+
+		setIsVerifying(true);
+		try {
+			const isValid = await verifyPassword(password);
+			if (!isValid) {
+				toast.error("Incorrect password");
+				return;
+			}
```

Comment:
This doesn't protect the endpoint, anyone could still call the performExport without going through the verify password. You should verify the password in the same backend function that does the export (ie: asks for the password in the performExport body)